build.sbt

```diff
@@ -124,20 +133,26 @@ lazy val lintFlags = {
     "-encoding",
     "UTF-8", // arg for -encoding
     "-feature",
-    "-unchecked",
+    "-unchecked"
+  )
+
+  lazy val preScala3LintFlags = allVersionLintFlags ++ List(
```

Comment:
Like `crossScalaVersions`, this is another aspect where I think that extracting out common parts is starting to make the build too complex. With four Scala versions, each mixing different subsets of flags, I'd prefer if we took the cost of duplication and simply defined one `List` per major version.

Reply:
I agree.